python/ray/dag/compiled_dag_node.py

```diff
@@ -1231,6 +1237,7 @@ def _execute_until(
 
         Args:
             execution_index: The execution index to execute until.
+            timeout: The maximum time in seconds to wait for the result.
```

Comment:
Specify behavior for values -1 and 0 (assuming these mean infinite and don't block as in ray.get).